ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/LMDB.hs

```diff
@@ -306,24 +348,31 @@ initFromVals ::
   -> Db m l
   -> m ()
 initFromVals dbsSeq vals Db{..} = liftIO $ LMDB.readWriteTransaction dbEnv $
-  withDbSettingsRWMaybeNull dbSettings $ \case
+  withDbStateRWMaybeNull dbState $ \case
     Nothing -> zipLedgerTablesA initLMDBTable dbBackingTables vals $> ((), DbState{dbsSeq})
     Just _ -> Exn.throw $ DbErrStr "initFromVals: db already had state"
 
-initFromLMDBs :: (MonadIO m, IOLike m)
+initFromLMDBs ::
+     (MonadIO m, IOLike m)
   => Trace.Tracer m TraceDb
+  -> LMDBLimits
   -> FS.SomeHasFS m
   -> FS.FsPath
   -> FS.FsPath
   -> m ()
-initFromLMDBs tracer shfs from0 to0 = do
-  Trace.traceWith tracer $ TDBInitialisingFromLMDB from0
-  from <- guardDbDir GDDMustExist shfs from0
-  to <- guardDbDir GDDMustNotExist shfs to0
-  bracket
-    (liftIO $ LMDB.openEnvironment from defaultLMDBLimits) -- TODO assess limits, in particular this could fail if the db is too big
-    (liftIO . LMDB.closeEnvironment)
-    (flip (lmdbCopy tracer) to)
+initFromLMDBs tracer limits shfs from0 to0 = do
+    Trace.traceWith tracer $ TDBInitialisingFromLMDB from0
+    from <- guardDbDir GDDMustExist shfs from0
+    to <- guardDbDirWithRetry GDDMustNotExist shfs to0
+    bracket
+      -- TODO assess limits, in particular this could fail if the db is too big
+      -- TODO(jdral): Should we read from some configuration/settings file what
+      -- the limits of the "from" database are? Do we want to support
+      -- interoperablity between databases with different limits?
```

Comment:
This requires a strategy.

Reply:
Don't we assume the limits to be given at this point? And isn't this a problem to be solved at the node?

Reply:
We assume that the limits are given to us.